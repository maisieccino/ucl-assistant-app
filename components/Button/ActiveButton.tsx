import { LinearGradient } from "expo-linear-gradient"
import _ from "lodash"
import React, { useCallback } from "react"
import {
  GestureResponderEvent,
  Platform,
  TouchableNativeFeedback,
  TouchableOpacity,
  ViewProps,
  ViewStyle,
} from "react-native"
import Colors from "../../constants/Colors"
import Styles from "../../styles/Button"

interface WrapperProps {
  children: React.ReactNode,
  disabled?: boolean,
  loading?: boolean,
  onPress: (event: GestureResponderEvent) => void,
  style?: ViewStyle,
}

const Wrapper: React.FC<WrapperProps> = ({ children, onPress, disabled }) => {
  const debouncedOnPress = useCallback(_.debounce((event: GestureResponderEvent) => {
    onPress(event)
  }, 200), [onPress])
  if (Platform.OS === `android`) {
    return (
      <TouchableNativeFeedback
        background={
          TouchableNativeFeedback.Ripple(Colors.accentColorLight, true)
        }
        onPress={debouncedOnPress}
        useForeground
        style={Styles.buttonWrapper}
        disabled={disabled}
      >
        {children}
      </TouchableNativeFeedback>
    )
  }
  if (Platform.OS === `ios`) {
    return (
      <TouchableOpacity
        style={{ backgroundColor: `transparent` }}
        onPress={onPress}
        disabled={disabled}
      >
        {children}
      </TouchableOpacity>
    )
  }
  return null
}

export interface ActiveButtonProps extends ViewProps {
  onPress?: (event: GestureResponderEvent) => void,
  style?: ViewStyle,
  children: React.ReactNode,
  disabled?: boolean,
}

const ActiveButton: React.FC<ActiveButtonProps> = ({
  onPress, style, children, disabled,
}) => (
    <Wrapper onPress={onPress} disabled={disabled}>
      <LinearGradient
        colors={[Colors.accentColor, Colors.buttonBackground]}
        style={[Styles.button, style]}
        start={[0, 1]}
        end={[1, 0]}
      >
        {children}
      </LinearGradient>
    </Wrapper>
)

export default ActiveButton
