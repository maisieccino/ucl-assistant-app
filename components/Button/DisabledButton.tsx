import React from "react"
import { TouchableNativeFeedback, View, ViewStyle } from "react-native"

import Styles from "../../styles/Button"

interface DisabledButtonProps {
  style?: ViewStyle,
  children: React.ReactNode,
}

const DisabledButton: React.FC<DisabledButtonProps> = ({
  style,
  children,
}) => (
    <TouchableNativeFeedback
      background={TouchableNativeFeedback.SelectableBackground()}
      useForeground
      style={Styles.buttonWrapper}
      disabled
    >
      <View style={[Styles.button, Styles.disabled, style]}>{children}</View>
    </TouchableNativeFeedback>
)

export default DisabledButton
