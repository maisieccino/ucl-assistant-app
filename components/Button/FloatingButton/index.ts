export { default, FloatingButtonProps } from "./FloatingButton"
