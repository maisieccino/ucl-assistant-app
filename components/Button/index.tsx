
export { default } from './Button'
export { default as RoundButton } from "./RoundButton"
export {
  default as FloatingButton,
  FloatingButtonProps,
} from "./FloatingButton"
export { default as LightButton } from './LightButton'
export { default as SmallButton } from './SmallButton'
