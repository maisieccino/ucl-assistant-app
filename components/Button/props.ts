import PropTypes from "prop-types"
import { ViewPropTypes } from "react-native"

export const propTypes = {
  children: PropTypes.node,
  disabled: PropTypes.bool,
  loading: PropTypes.bool,
  onPress: PropTypes.func,
  style: ViewPropTypes.style,
}

export const defaultProps = {
  children: ``,
  disabled: false,
  loading: false,
  onPress: () => { },
  style: {},
}
