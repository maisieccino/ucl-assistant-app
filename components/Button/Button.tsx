import React from "react"
import { ActivityIndicator, Platform } from "react-native"

import Colors from "../../constants/Colors"
import { ButtonText } from "../Typography"
import ActiveButton, { ActiveButtonProps } from "./ActiveButton"
import DisabledButton from "./DisabledButton"

export interface ButtonProps extends ActiveButtonProps {
  loading?: boolean,
}

const Button: React.FC<ButtonProps> = ({
  loading = false,
  disabled = false,
  children,
  ...otherProps
}) => {
  const buttonSize = Platform.OS === `android` ? 24 : 1
  let innerComponent = children
  if (typeof children === `string`) {
    innerComponent = (
      <ButtonText>
        {children}
      </ButtonText>
    )
  }
  if (loading) {
    innerComponent = (
      <ActivityIndicator size={buttonSize} color={Colors.pageBackground} />
    )
  }
  if (disabled) {
    return (
      <DisabledButton {...otherProps}>
        {innerComponent}
      </DisabledButton>
    )
  }
  return (
    <ActiveButton {...otherProps}>
      {innerComponent}
    </ActiveButton>
  )
}

export default Button
