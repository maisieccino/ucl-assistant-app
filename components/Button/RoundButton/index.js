import { Feather } from "@expo/vector-icons"
import React from "react"
import { ActivityIndicator, Platform } from "react-native"

import Colors from "../../../constants/Colors"
import ActiveButton from "./ActiveButton"
import DisabledButton from "./DisabledButton"
import { defaultProps, propTypes } from "./props"

const RoundButton = (props) => {
  const { icon, loading, disabled } = props
  const buttonSize = Platform.OS === `android` ? 24 : 16
  let child = (
    <Feather size={buttonSize} name={icon} color={Colors.pageBackground} />
  )
  if (loading) {
    child = (
      <ActivityIndicator size={buttonSize} color={Colors.pageBackground} />
    )
  }
  if (disabled) {
    return <DisabledButton {...props}>{child}</DisabledButton>
  }
  return <ActiveButton {...props}>{child}</ActiveButton>
}

RoundButton.propTypes = propTypes
RoundButton.defaultProps = defaultProps

export default RoundButton
