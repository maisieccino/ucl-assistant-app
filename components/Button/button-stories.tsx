import { action } from "@storybook/addon-actions"
import { boolean, text, withKnobs } from "@storybook/addon-knobs"
import { storiesOf } from "@storybook/react-native"
import React from "react"

import CenterView from "../CenterView"
import Button from "."
import RoundButton from "./RoundButton"

storiesOf(`Button/Default`, module)
  .addDecorator((getStory) => <CenterView>{getStory()}</CenterView>)
  .addDecorator(withKnobs)
  .add(`with text`, () => (
    <Button
      onPress={action(`clicked-text`)}
      disabled={boolean(`Disabled`, false)}
      loading={boolean(`Loading`, false)}
    >
      {text(`Text`, `Hello Button`)}
    </Button>
  ))
  .add(`with some emoji`, () => (
    <Button onPress={action(`clicked-emoji`)}>😀 😎 👍 💯</Button>
  ))
  .add(`with disabled prop`, () => (
    <Button disabled onPress={action(`clicked-disabled`)}>
      Disabled Button
    </Button>
  ))

storiesOf(`Button/Round`, module)
  .addDecorator((getStory) => <CenterView>{getStory()}</CenterView>)
  .addDecorator(withKnobs)
  .add(`with icon`, () => (
    <RoundButton
      onPress={action(`clicked-round`)}
      disabled={boolean(`Disabled`, false)}
      loading={boolean(`Loading`, false)}
    />
  ))
  .add(`with disabled prop`, () => (
    <RoundButton
      onPress={action(`clicked-round`)}
      loading={boolean(`Loading`, false)}
      disabled
    />
  ))
