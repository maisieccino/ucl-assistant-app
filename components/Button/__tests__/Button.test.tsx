import { cleanup, fireEvent, render } from "@testing-library/react-native"
import React from 'react'
import "react-native"
import Button from ".."
import { ButtonText } from '../../Typography'

describe(`Button`, () => {
  beforeEach(() => {
    jest.useRealTimers()
    jest.clearAllMocks()
  })
  afterEach(() => {
    cleanup()
  })
  it(`renders with raw text as child`, () => {
    const wrapper = render(
      <Button>
        Press Me
      </Button>,
    )
    expect(wrapper).toMatchSnapshot()
  })

  it(`renders with text element as child`, () => {
    const wrapper = render(
      <Button>
        <ButtonText>Press Me</ButtonText>
      </Button>,
    )
    expect(wrapper).toMatchSnapshot()
  })

  it(`renders the loading state`, () => {
    const wrapper = render(
      <Button loading>
        This is a loading button
      </Button>,
    )
    expect(wrapper).toMatchSnapshot()
  })

  it(`handles onPress events`, () => {
    const onPress = jest.fn()
    const wrapper = render(
      <Button
        onPress={onPress}
      >
        Active Button
      </Button>,
    )

    const { getByText } = wrapper
    const activeButton = getByText(/Active Button/i)
    fireEvent.press(activeButton)

    expect(onPress).toHaveBeenCalledTimes(1)
    expect(onPress.mock.calls).toMatchSnapshot()
  })

  it(`renders the disabled state`, () => {
    const onPress = jest.fn()
    const wrapper = render(
      <Button
        disabled
        onPress={onPress}
      >
        This is a disabled button
      </Button>,
    )
    expect(wrapper).toMatchSnapshot()

    const { getByText } = wrapper
    const disabledButton = getByText(/disabled button/i)
    fireEvent.press(disabledButton)

    expect(onPress).toMatchSnapshot()
    expect(onPress).toHaveBeenCalledTimes(0)
  })
})
