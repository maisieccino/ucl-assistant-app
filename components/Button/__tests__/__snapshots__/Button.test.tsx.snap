// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button handles onPress events 1`] = `
Array [
  Array [],
]
`;

exports[`Button renders the disabled state 1`] = `
Object {
  "asJSON": [Function],
  "baseElement": <View
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      collapsable={true}
      pointerEvents="box-none"
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <TouchableNativeFeedback
        background={
          Object {
            "attribute": "selectableItemBackground",
            "type": "ThemeAttrAndroid",
          }
        }
        disabled={true}
        style={
          Object {
            "borderRadius": 8,
            "flex": 1,
            "justifyContent": "center",
          }
        }
        useForeground={true}
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "borderRadius": 8,
                "elevation": 2,
                "flexDirection": "row",
                "justifyContent": "center",
                "marginBottom": 5,
                "marginTop": 5,
                "minHeight": 55,
                "minWidth": 150,
                "padding": 8,
                "paddingLeft": 16,
                "paddingRight": 16,
                "shadowColor": "#1B998B",
                "shadowOffset": Object {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 0.75,
                "shadowRadius": 3,
              },
              Object {
                "backgroundColor": "rgba(231, 231, 233, 1)",
              },
              undefined,
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#fefefe",
                  "fontFamily": "apercu",
                  "fontSize": 20,
                  "marginBottom": 1,
                  "marginTop": 1,
                  "textAlign": "center",
                },
                undefined,
              ]
            }
          >
            This is a disabled button
          </Text>
        </View>
      </TouchableNativeFeedback>
    </View>
  </View>,
  "container": <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <TouchableNativeFeedback
      background={
        Object {
          "attribute": "selectableItemBackground",
          "type": "ThemeAttrAndroid",
        }
      }
      disabled={true}
      style={
        Object {
          "borderRadius": 8,
          "flex": 1,
          "justifyContent": "center",
        }
      }
      useForeground={true}
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "borderRadius": 8,
              "elevation": 2,
              "flexDirection": "row",
              "justifyContent": "center",
              "marginBottom": 5,
              "marginTop": 5,
              "minHeight": 55,
              "minWidth": 150,
              "padding": 8,
              "paddingLeft": 16,
              "paddingRight": 16,
              "shadowColor": "#1B998B",
              "shadowOffset": Object {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.75,
              "shadowRadius": 3,
            },
            Object {
              "backgroundColor": "rgba(231, 231, 233, 1)",
            },
            undefined,
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#fefefe",
                "fontFamily": "apercu",
                "fontSize": 20,
                "marginBottom": 1,
                "marginTop": 1,
                "textAlign": "center",
              },
              undefined,
            ]
          }
        >
          This is a disabled button
        </Text>
      </View>
    </TouchableNativeFeedback>
  </View>,
  "debug": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByHintText": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByDisplayValue": [Function],
  "findByHintText": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByHintText": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByDisplayValue": [Function],
  "getByHintText": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByHintText": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByDisplayValue": [Function],
  "queryByHintText": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Button renders the disabled state 2`] = `[MockFunction]`;

exports[`Button renders the loading state 1`] = `
Object {
  "asJSON": [Function],
  "baseElement": <View
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      collapsable={true}
      pointerEvents="box-none"
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <TouchableOpacity
        style={
          Object {
            "backgroundColor": "transparent",
          }
        }
      >
        <ViewManagerAdapter_ExpoLinearGradient
          proxiedProperties={
            Object {
              "colors": Array [
                4279998859,
                4283151357,
              ],
              "endPoint": Array [
                1,
                0,
              ],
              "locations": undefined,
              "startPoint": Array [
                0,
                1,
              ],
            }
          }
          style={
            Array [
              Object {
                "alignItems": "center",
                "borderRadius": 8,
                "elevation": 2,
                "flexDirection": "row",
                "justifyContent": "center",
                "marginBottom": 5,
                "marginTop": 5,
                "minHeight": 55,
                "minWidth": 150,
                "padding": 8,
                "paddingLeft": 16,
                "paddingRight": 16,
                "shadowColor": "#1B998B",
                "shadowOffset": Object {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 0.75,
                "shadowRadius": 3,
              },
              undefined,
            ]
          }
        >
          <ActivityIndicator
            animating={true}
            color="#fefefe"
            hidesWhenStopped={true}
            size={1}
          />
        </ViewManagerAdapter_ExpoLinearGradient>
      </TouchableOpacity>
    </View>
  </View>,
  "container": <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <TouchableOpacity
      style={
        Object {
          "backgroundColor": "transparent",
        }
      }
    >
      <ViewManagerAdapter_ExpoLinearGradient
        proxiedProperties={
          Object {
            "colors": Array [
              4279998859,
              4283151357,
            ],
            "endPoint": Array [
              1,
              0,
            ],
            "locations": undefined,
            "startPoint": Array [
              0,
              1,
            ],
          }
        }
        style={
          Array [
            Object {
              "alignItems": "center",
              "borderRadius": 8,
              "elevation": 2,
              "flexDirection": "row",
              "justifyContent": "center",
              "marginBottom": 5,
              "marginTop": 5,
              "minHeight": 55,
              "minWidth": 150,
              "padding": 8,
              "paddingLeft": 16,
              "paddingRight": 16,
              "shadowColor": "#1B998B",
              "shadowOffset": Object {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.75,
              "shadowRadius": 3,
            },
            undefined,
          ]
        }
      >
        <ActivityIndicator
          animating={true}
          color="#fefefe"
          hidesWhenStopped={true}
          size={1}
        />
      </ViewManagerAdapter_ExpoLinearGradient>
    </TouchableOpacity>
  </View>,
  "debug": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByHintText": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByDisplayValue": [Function],
  "findByHintText": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByHintText": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByDisplayValue": [Function],
  "getByHintText": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByHintText": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByDisplayValue": [Function],
  "queryByHintText": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Button renders with raw text as child 1`] = `
Object {
  "asJSON": [Function],
  "baseElement": <View
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      collapsable={true}
      pointerEvents="box-none"
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <TouchableOpacity
        style={
          Object {
            "backgroundColor": "transparent",
          }
        }
      >
        <ViewManagerAdapter_ExpoLinearGradient
          proxiedProperties={
            Object {
              "colors": Array [
                4279998859,
                4283151357,
              ],
              "endPoint": Array [
                1,
                0,
              ],
              "locations": undefined,
              "startPoint": Array [
                0,
                1,
              ],
            }
          }
          style={
            Array [
              Object {
                "alignItems": "center",
                "borderRadius": 8,
                "elevation": 2,
                "flexDirection": "row",
                "justifyContent": "center",
                "marginBottom": 5,
                "marginTop": 5,
                "minHeight": 55,
                "minWidth": 150,
                "padding": 8,
                "paddingLeft": 16,
                "paddingRight": 16,
                "shadowColor": "#1B998B",
                "shadowOffset": Object {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 0.75,
                "shadowRadius": 3,
              },
              undefined,
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#fefefe",
                  "fontFamily": "apercu",
                  "fontSize": 20,
                  "marginBottom": 1,
                  "marginTop": 1,
                  "textAlign": "center",
                },
                undefined,
              ]
            }
          >
            Press Me
          </Text>
        </ViewManagerAdapter_ExpoLinearGradient>
      </TouchableOpacity>
    </View>
  </View>,
  "container": <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <TouchableOpacity
      style={
        Object {
          "backgroundColor": "transparent",
        }
      }
    >
      <ViewManagerAdapter_ExpoLinearGradient
        proxiedProperties={
          Object {
            "colors": Array [
              4279998859,
              4283151357,
            ],
            "endPoint": Array [
              1,
              0,
            ],
            "locations": undefined,
            "startPoint": Array [
              0,
              1,
            ],
          }
        }
        style={
          Array [
            Object {
              "alignItems": "center",
              "borderRadius": 8,
              "elevation": 2,
              "flexDirection": "row",
              "justifyContent": "center",
              "marginBottom": 5,
              "marginTop": 5,
              "minHeight": 55,
              "minWidth": 150,
              "padding": 8,
              "paddingLeft": 16,
              "paddingRight": 16,
              "shadowColor": "#1B998B",
              "shadowOffset": Object {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.75,
              "shadowRadius": 3,
            },
            undefined,
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#fefefe",
                "fontFamily": "apercu",
                "fontSize": 20,
                "marginBottom": 1,
                "marginTop": 1,
                "textAlign": "center",
              },
              undefined,
            ]
          }
        >
          Press Me
        </Text>
      </ViewManagerAdapter_ExpoLinearGradient>
    </TouchableOpacity>
  </View>,
  "debug": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByHintText": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByDisplayValue": [Function],
  "findByHintText": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByHintText": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByDisplayValue": [Function],
  "getByHintText": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByHintText": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByDisplayValue": [Function],
  "queryByHintText": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Button renders with text element as child 1`] = `
Object {
  "asJSON": [Function],
  "baseElement": <View
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      collapsable={true}
      pointerEvents="box-none"
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <TouchableOpacity
        style={
          Object {
            "backgroundColor": "transparent",
          }
        }
      >
        <ViewManagerAdapter_ExpoLinearGradient
          proxiedProperties={
            Object {
              "colors": Array [
                4279998859,
                4283151357,
              ],
              "endPoint": Array [
                1,
                0,
              ],
              "locations": undefined,
              "startPoint": Array [
                0,
                1,
              ],
            }
          }
          style={
            Array [
              Object {
                "alignItems": "center",
                "borderRadius": 8,
                "elevation": 2,
                "flexDirection": "row",
                "justifyContent": "center",
                "marginBottom": 5,
                "marginTop": 5,
                "minHeight": 55,
                "minWidth": 150,
                "padding": 8,
                "paddingLeft": 16,
                "paddingRight": 16,
                "shadowColor": "#1B998B",
                "shadowOffset": Object {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 0.75,
                "shadowRadius": 3,
              },
              undefined,
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#fefefe",
                  "fontFamily": "apercu",
                  "fontSize": 20,
                  "marginBottom": 1,
                  "marginTop": 1,
                  "textAlign": "center",
                },
                undefined,
              ]
            }
          >
            Press Me
          </Text>
        </ViewManagerAdapter_ExpoLinearGradient>
      </TouchableOpacity>
    </View>
  </View>,
  "container": <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <TouchableOpacity
      style={
        Object {
          "backgroundColor": "transparent",
        }
      }
    >
      <ViewManagerAdapter_ExpoLinearGradient
        proxiedProperties={
          Object {
            "colors": Array [
              4279998859,
              4283151357,
            ],
            "endPoint": Array [
              1,
              0,
            ],
            "locations": undefined,
            "startPoint": Array [
              0,
              1,
            ],
          }
        }
        style={
          Array [
            Object {
              "alignItems": "center",
              "borderRadius": 8,
              "elevation": 2,
              "flexDirection": "row",
              "justifyContent": "center",
              "marginBottom": 5,
              "marginTop": 5,
              "minHeight": 55,
              "minWidth": 150,
              "padding": 8,
              "paddingLeft": 16,
              "paddingRight": 16,
              "shadowColor": "#1B998B",
              "shadowOffset": Object {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.75,
              "shadowRadius": 3,
            },
            undefined,
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#fefefe",
                "fontFamily": "apercu",
                "fontSize": 20,
                "marginBottom": 1,
                "marginTop": 1,
                "textAlign": "center",
              },
              undefined,
            ]
          }
        >
          Press Me
        </Text>
      </ViewManagerAdapter_ExpoLinearGradient>
    </TouchableOpacity>
  </View>,
  "debug": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByHintText": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByDisplayValue": [Function],
  "findByHintText": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByHintText": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByDisplayValue": [Function],
  "getByHintText": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByHintText": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByDisplayValue": [Function],
  "queryByHintText": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
